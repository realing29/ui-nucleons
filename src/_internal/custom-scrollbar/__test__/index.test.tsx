import { mount } from 'enzyme';
import React from 'react';
import { CustomScrollbar, HandleFullScroll } from '..';

describe('CustomScrollbar', () => {
  it('should handle onFullScroll/fullScrollThreshold', () => {
    const spy = jest.fn();

    const wrapper = mount(<CustomScrollbar onFullScroll={spy} fullScrollThreshold={120} />);

    expect(wrapper).toMatchSnapshot();
  });
});

describe('HandleFullScroll', () => {
  it('should handle event', () => {
    const spy = jest.fn();
    const handler = HandleFullScroll(spy, 10);

    expect(spy).toHaveBeenCalledTimes(0);

    handler({
      target: {
        __proto__: Element.prototype,
        clientHeight: 100,
        scrollTop: 0,
        scrollHeight: 1000,
      },
    } as any);

    expect(spy).toHaveBeenCalledTimes(0);

    handler({
      target: {
        __proto__: Element.prototype,
        clientHeight: 100,
        scrollTop: 900,
        scrollHeight: 1000,
      },
    } as any);

    expect(spy).toHaveBeenCalledTimes(1);
  });

  it('should handle undefined', () => {
    const spy = jest.fn();
    const handler = HandleFullScroll(spy);

    expect(spy).toHaveBeenCalledTimes(0);

    handler(undefined);

    expect(spy).toHaveBeenCalledTimes(0);
  });
});
