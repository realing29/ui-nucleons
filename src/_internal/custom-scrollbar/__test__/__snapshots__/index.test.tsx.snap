// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomScrollbar should handle onFullScroll/fullScrollThreshold 1`] = `
<CustomScrollbar
  fullScrollThreshold={120}
  onFullScroll={[MockFunction]}
>
  <OverlayScrollbarsComponent
    className="custom-scrollbar"
    options={
      Object {
        "callbacks": Object {
          "onScrollStop": [Function],
        },
        "overflowBehavior": undefined,
      }
    }
  >
    <div
      className="os-host"
    >
      <div
        className="os-resize-observer-host"
      />
      <div
        className="os-padding"
      >
        <div
          className="os-viewport"
        >
          <div
            className="os-content"
          />
        </div>
      </div>
      <div
        className="os-scrollbar os-scrollbar-horizontal "
      >
        <div
          className="os-scrollbar-track"
        >
          <div
            className="os-scrollbar-handle"
          />
        </div>
      </div>
      <div
        className="os-scrollbar os-scrollbar-vertical"
      >
        <div
          className="os-scrollbar-track"
        >
          <div
            className="os-scrollbar-handle"
          />
        </div>
      </div>
      <div
        className="os-scrollbar-corner"
      />
    </div>
  </OverlayScrollbarsComponent>
</CustomScrollbar>
`;
